### PR TITLE
added UI output adapter

### PR DESCRIPTION
**Pull request by ayyoob**

### CODE REVIEW

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/SuperTenantSubscriptionEndpoint.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+				ServiceHolder.getInstance().getUiOutputCallbackControllerService().subscribeWebsocket(streamName,
+																									  version, session);
+			} finally {
+				PrivilegedCarbonContext.endTenantFlow();
+			}
+		} else {
+			try {
+				session.close(new CloseReason(CloseReason.CloseCodes.CANNOT_ACCEPT, "Unauthorized Access"));
+			} catch (IOException e) {
+				log.error("Failed to disconnect the unauthorized client.");
```
Comment (prabathabey) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/SuperTenantSubscriptionEndpoint.java]: Exception trace is not logged. Please fix.

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/TenantSubscriptionEndpoint.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+                ServiceHolder.getInstance().getUiOutputCallbackControllerService().subscribeWebsocket(streamName,
+                                                                                                      version, session);
+            } finally {
+                PrivilegedCarbonContext.endTenantFlow();
+            }
+        } else {
+            try {
+                session.close(new CloseReason(CloseReason.CloseCodes.CANNOT_ACCEPT, "Unauthorized Access"));
+            } catch (IOException e) {
+                log.error("Failed to disconnect the unauthorized client.");
```
Comment (prabathabey) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/TenantSubscriptionEndpoint.java]: Exception trace is not logged here as well.

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/TenantSubscriptionEndpoint.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+     * Web socket onMessage - When client sens a message
+     *
+     * @param session - Users registered session.
+     * @param message  - Status code for web-socket close.
+     * @param streamName - StreamName extracted from the ws url.
+     */
+    @OnMessage
+    public void onMessage (Session session, String message, @PathParam("streamname") String streamName, @PathParam("tdomain") String tdomain) {
+        if (log.isDebugEnabled()) {
+            log.debug("Received and dropped message from client. Message: " + message+", for Session id: "+session.getId()+", for tenant domain"+tdomain+", for the Adaptor:"+streamName);
```
Comment (prabathabey) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/TenantSubscriptionEndpoint.java]: This line is not properly formatted.

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/oauth/OAuthTokenValdiator.java [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+		}
+		return  oAuthTokenValdiator;
+	}
+
+	private OAuthTokenValdiator() {
+		try {
+			Properties properties = getWebSocketConfig();
+			this.stubs = new GenericObjectPool(new OAuthTokenValidaterStubFactory(properties));
+		} catch (IOException e) {
+			log.error("Failed to parse the web socket config file " + WEBSOCKET_CONFIG_LOCATION);
```
Comment (prabathabey) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/oauth/OAuthTokenValdiator.java]: Exception trace is not logged here as well.

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/oauth/OAuthTokenValdiator.java [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+	}
+
+	/**
+	 * Retrieve JWT configs from registry.
+	 */
+	private Properties getWebSocketConfig() throws IOException {
+		Properties properties = new Properties();
+		File configFile =new File(WEBSOCKET_CONFIG_LOCATION);
+		if (configFile.exists()) {
+			InputStream fileInputStream = new FileInputStream(configFile);
```
Comment (prabathabey) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/oauth/OAuthTokenValdiator.java]: Input stream is not closed explicitly, which can lead to possible OOM issues.

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/oauth/OAuthTokenValidaterStubFactory.java [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.security.GeneralSecurityException;
+import java.util.Properties;
+
+/**
+ * This follows object pool pattern to manage the stub for oauth validation service.
+ */
+public class OAuthTokenValidaterStubFactory extends BasePoolableObjectFactory {
+	private static final Logger log = Logger.getLogger(OAuthTokenValidaterStubFactory.class);
```
Comment (prabathabey) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/oauth/OAuthTokenValidaterStubFactory.java]: Would have been better to stick to commons logging standards for logging so you can change the underlying implementation to any commons logging supported logging scheme without having to modify the code.

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/oauth/exception/OAuthTokenValidationException.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package oauth.exception;
+
+/**
+ * This Exception will be thrown, when there any interference with token validation flow.
+ */
+public class OAuthTokenValidationException extends Exception {
+	private String errMessage;
```
Comment (prabathabey) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/oauth/exception/OAuthTokenValidationException.java]: Redundant caching of the error message. Would be a good idea to remove it.

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/util/ServiceHolder.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    private static ServiceHolder instance;
+    private UIOutputCallbackControllerService uiOutputCallbackControllerService;
+
+    private ServiceHolder(){
+        uiOutputCallbackControllerService = (UIOutputCallbackControllerService) PrivilegedCarbonContext
+                .getThreadLocalCarbonContext().getOSGiService(UIOutputCallbackControllerService.class, null);
+    }
+
+    public synchronized static ServiceHolder getInstance(){
+        if (instance==null){
```
Comment (prabathabey) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/util/ServiceHolder.java]: This code block is not properly formatted.

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/util/UIConstants.java [hunk 9]
```diff
@@ -0,0 +1,38 @@
+/*
```
Comment (prabathabey) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/util/UIConstants.java]: Remove unnecessary newlines.

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/util/UIConstants.java [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+ *  *
+ *
+ */
+
+package util;
+
+/**
+ * This class contains the constants related to ui client.
+ */
+public class UIConstants {
```
Comment (prabathabey) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/util/UIConstants.java]: Might be elegant to declare the class "final".

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/util/UIConstants.java [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+    private UIConstants() {
```
Comment (prabathabey) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/util/UIConstants.java]: Properly format the code implemented within the class body.

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/UIEventAdapter.java [hunk 12]
```diff
@@ -0,0 +1,455 @@
...
+    public UIEventAdapter(OutputEventAdapterConfiguration eventAdapterConfiguration, Map<String,
+            String> globalProperties) {
+        this.eventAdapterConfiguration = eventAdapterConfiguration;
+        this.globalProperties = globalProperties;
+        this.doLogDroppedMessage = true;
+    }
+
+    @Override
+    public void init() throws OutputEventAdapterException {
+
```
Comment (prabathabey) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/UIEventAdapter.java]: Unnecessary new line.

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/UIEventAdapter.java [hunk 13]
```diff
@@ -0,0 +1,455 @@
...
+    }
+
+    @Override
+    public void connect() {
+        //Not needed
+    }
+
+    @Override
+    public void publish(Object message, Map<String, String> dynamicProperties) {
+
```
Comment (prabathabey) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/UIEventAdapter.java]: Unnecessary new line.

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/UIEventAdapter.java [hunk 14]
```diff
@@ -0,0 +1,455 @@
...
+    }
+
+    @Override
+    public void disconnect() {
+        //Not needed
+    }
+
+    @Override
+    public void destroy() {
+
```
Comment (prabathabey) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/UIEventAdapter.java]: Unnecessary new line.

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/UIEventAdapterFactory.java [hunk 15]
```diff
@@ -0,0 +1,81 @@
+/*
```
Comment (prabathabey) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/UIEventAdapterFactory.java]: Remove unnecessary new line.

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/internal/UIEventAdaptorServiceDataHolder.java [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+import java.util.concurrent.LinkedBlockingDeque;
+
+/**
+ * Creates a holder of type UIOutputCallbackRegisterServiceImpl.
+ */
+public final class UIEventAdaptorServiceDataHolder {
+
+    private static UIOutputCallbackControllerServiceImpl UIOutputCallbackRegisterServiceImpl;
+    private static ConcurrentHashMap<Integer,ConcurrentHashMap<String, String>>
+            tenantSpecificOutputEventStreamAdapterMap = new ConcurrentHashMap<Integer,ConcurrentHashMap<String, String>>();
```
Comment (prabathabey) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/internal/UIEventAdaptorServiceDataHolder.java]: This line is not properly formatted.

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/internal/UIEventAdaptorServiceDataHolder.java [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.LinkedBlockingDeque;
+
+/**
+ * Creates a holder of type UIOutputCallbackRegisterServiceImpl.
+ */
+public final class UIEventAdaptorServiceDataHolder {
+
+    private static UIOutputCallbackControllerServiceImpl UIOutputCallbackRegisterServiceImpl;
+    private static ConcurrentHashMap<Integer,ConcurrentHashMap<String, String>>
```
Comment (prabathabey) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/internal/UIEventAdaptorServiceDataHolder.java]: This line is not properly formatted.

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/internal/UIEventAdaptorServiceDataHolder.java [hunk 18]
```diff
@@ -0,0 +1,67 @@
...
+            tenantSpecificOutputEventStreamAdapterMap = new ConcurrentHashMap<Integer,ConcurrentHashMap<String, String>>();
+    private static ConcurrentHashMap<Integer, ConcurrentHashMap<String, LinkedBlockingDeque<Object>>>
+            tenantSpecificStreamEventMap = new ConcurrentHashMap<Integer, ConcurrentHashMap<String, LinkedBlockingDeque<Object>>>();
```
Comment (prabathabey) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/internal/UIEventAdaptorServiceDataHolder.java]: So is this.

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/internal/UILocalEventAdapterServiceComponent.java [hunk 19]
```diff
@@ -0,0 +1,79 @@
...
+                    new UIOutputCallbackControllerServiceImpl();
+            context.getBundleContext().registerService(UIOutputCallbackControllerService.class.getName(),
+                    UIOutputCallbackRegisterServiceImpl, null);
+
+            UIEventAdaptorServiceDataHolder.registerUIOutputCallbackRegisterServiceInternal(
+                    UIOutputCallbackRegisterServiceImpl);
+            if (log.isDebugEnabled()) {
+                log.debug("Successfully deployed the output ui adapter service");
+            }
+        } catch (RuntimeException e) {
```
Comment (prabathabey) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/internal/UILocalEventAdapterServiceComponent.java]: Catch throwable and log. Ideally, any error thrown is likely to halt bundle start up.

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/SuperTenantSubscriptionEndpoint.java (hunk 1 above)
Comment (ayyoob) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/SuperTenantSubscriptionEndpoint.java]: Fixed with https://github.com/wso2/carbon-device-mgt-plugins/pull/309/commits/20cf418394c09a4cd473e8f2cc05746e3a9e8f36

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/TenantSubscriptionEndpoint.java (hunk 2 above)
Comment (ayyoob) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/TenantSubscriptionEndpoint.java]: Fixed with https://github.com/wso2/carbon-device-mgt-plugins/pull/309/commits/20cf418394c09a4cd473e8f2cc05746e3a9e8f36

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/TenantSubscriptionEndpoint.java (hunk 3 above)
Comment (ayyoob) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/TenantSubscriptionEndpoint.java]: Fixed with https://github.com/wso2/carbon-device-mgt-plugins/pull/309/commits/20cf418394c09a4cd473e8f2cc05746e3a9e8f36

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/oauth/OAuthTokenValdiator.java (hunk 4 above)
Comment (ayyoob) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/oauth/OAuthTokenValdiator.java]: Fixed with https://github.com/wso2/carbon-device-mgt-plugins/pull/309/commits/20cf418394c09a4cd473e8f2cc05746e3a9e8f36

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/oauth/OAuthTokenValdiator.java (hunk 5 above)
Comment (ayyoob) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/oauth/OAuthTokenValdiator.java]: This code is already been refactored in the master

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/oauth/OAuthTokenValidaterStubFactory.java (hunk 6 above)
Comment (ayyoob) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/oauth/OAuthTokenValidaterStubFactory.java]: Fixed with https://github.com/wso2/carbon-device-mgt-plugins/pull/309/commits/20cf418394c09a4cd473e8f2cc05746e3a9e8f36

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/oauth/exception/OAuthTokenValidationException.java (hunk 7 above)
Comment (ayyoob) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/oauth/exception/OAuthTokenValidationException.java]: Fixed with https://github.com/wso2/carbon-device-mgt-plugins/pull/310/commits/b60487304f8a05378e89c03a96aa893347e29e68

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/util/ServiceHolder.java (hunk 8 above)
Comment (ayyoob) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/util/ServiceHolder.java]: Fixed with https://github.com/wso2/carbon-device-mgt-plugins/pull/309/commits/20cf418394c09a4cd473e8f2cc05746e3a9e8f36

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/util/UIConstants.java (hunk 9 above)
Comment (ayyoob) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/util/UIConstants.java]: Fixed with https://github.com/wso2/carbon-device-mgt-plugins/pull/309/commits/20cf418394c09a4cd473e8f2cc05746e3a9e8f36

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/util/UIConstants.java (hunk 10 above)
Comment (ayyoob) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/util/UIConstants.java]: Fixed with https://github.com/wso2/carbon-device-mgt-plugins/pull/309/commits/20cf418394c09a4cd473e8f2cc05746e3a9e8f36

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/util/UIConstants.java (hunk 11 above)
Comment (ayyoob) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/util/UIConstants.java]: Fixed with https://github.com/wso2/carbon-device-mgt-plugins/pull/309/commits/20cf418394c09a4cd473e8f2cc05746e3a9e8f36

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/UIEventAdapter.java (hunk 12 above)
Comment (ayyoob) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/UIEventAdapter.java]: Fixed with https://github.com/wso2/carbon-device-mgt-plugins/pull/309/commits/20cf418394c09a4cd473e8f2cc05746e3a9e8f36

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/UIEventAdapter.java (hunk 13 above)
Comment (ayyoob) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/UIEventAdapter.java]: Fixed with https://github.com/wso2/carbon-device-mgt-plugins/pull/309/commits/20cf418394c09a4cd473e8f2cc05746e3a9e8f36

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/UIEventAdapter.java (hunk 14 above)
Comment (ayyoob) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/UIEventAdapter.java]: Fixed with https://github.com/wso2/carbon-device-mgt-plugins/pull/309/commits/20cf418394c09a4cd473e8f2cc05746e3a9e8f36

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/UIEventAdapterFactory.java (hunk 15 above)
Comment (ayyoob) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/UIEventAdapterFactory.java]: Fixed with https://github.com/wso2/carbon-device-mgt-plugins/pull/311/commits/618230e1215b4977ee9ad1f0f4f3279464be6918

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/internal/UIEventAdaptorServiceDataHolder.java (hunk 16 above)
Comment (ayyoob) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/internal/UIEventAdaptorServiceDataHolder.java]: fixed with https://github.com/wso2/carbon-device-mgt-plugins/pull/309/commits/20cf418394c09a4cd473e8f2cc05746e3a9e8f36

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/internal/UIEventAdaptorServiceDataHolder.java (hunk 17 above)
Comment (ayyoob) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/internal/UIEventAdaptorServiceDataHolder.java]: Fixed with https://github.com/wso2/carbon-device-mgt-plugins/pull/309/commits/20cf418394c09a4cd473e8f2cc05746e3a9e8f36

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/internal/UIEventAdaptorServiceDataHolder.java (hunk 18 above)
Comment (ayyoob) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/internal/UIEventAdaptorServiceDataHolder.java]: Fixed with https://github.com/wso2/carbon-device-mgt-plugins/pull/309/commits/20cf418394c09a4cd473e8f2cc05746e3a9e8f36

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/internal/UILocalEventAdapterServiceComponent.java (hunk 19 above)
Comment (ayyoob) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/internal/UILocalEventAdapterServiceComponent.java]: Fixed with https://github.com/wso2/carbon-device-mgt-plugins/pull/309/commits/20cf418394c09a4cd473e8f2cc05746e3a9e8f36

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/util/WebSocketSessionUtil.java [hunk 20]
```diff
@@ -0,0 +1,62 @@
+package org.wso2.carbon.device.mgt.iot.output.adapter.ui.util;
```
Comment (mohanvive) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/util/WebSocketSessionUtil.java]: where is the license header ?

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/oauth/OAuthTokenValidaterStubFactory.java [hunk 21]
```diff
@@ -0,0 +1,177 @@
...
+					options.setProperty(HTTPConstants.AUTHENTICATE, auth);
+					options.setProperty(HTTPConstants.REUSE_HTTP_CLIENT, Constants.VALUE_TRUE);
+					client.setOptions(options);
+					if (hostURL.getProtocol().equals("https")) {
+						// set up ssl factory since axis2 https transport is used.
+						EasySSLProtocolSocketFactory sslProtocolSocketFactory =
+								createProtocolSocketFactory();
+						Protocol authhttps = new Protocol(hostURL.getProtocol(),
+														  (ProtocolSocketFactory) sslProtocolSocketFactory,
+														  hostURL.getPort());
```
Comment (mohanvive) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui.endpoint/src/main/java/oauth/OAuthTokenValidaterStubFactory.java]: Code is not properly formatted.. Please fix that..

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/util/WebSocketSessionUtil.java [hunk 22]
```diff
@@ -0,0 +1,62 @@
...
+            for (String keyValuePair : allQueryParamPairs) {
+                String[] thisQueryParamPair = keyValuePair.split(QUERY_KEY_VALUE_SEPERATOR);
+
+                if (thisQueryParamPair.length != 2) {
+                    log.warn("Invalid query string [" + queryString + "] passed in.");
+                    break;
+                }
+
+                if (queryParamValuePairs == null) {
+                    queryParamValuePairs = new HashMap<>();
```
Comment (mohanvive) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/util/WebSocketSessionUtil.java]: Do we really need this if check? Can't we simply initialize the map with hashmap ?

File: components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/util/WebSocketSessionUtil.java (hunk 20 above)
Comment (ayyoob) [inline components/iot-plugins/iot-base-plugin/org.wso2.carbon.device.mgt.iot.output.adapter.ui/src/main/java/org/wso2/carbon/device/mgt/iot/output/adapter/ui/util/WebSocketSessionUtil.java]: Fixed with https://github.com/wso2/carbon-device-mgt-plugins/pull/385/commits/391cdbb43529ec1ec6971be70a3fdd7bc2b8789d

**Pull request merged**